### PR TITLE
Processing updates

### LINKED ISSUE #67: Need to save curTripleTerm on entry/exit to reifiedTriple and annotationBlock
Consider the following:

```turtle
:s :p :o ~:r1 { :p1 << :s2 :p2 :o2 >> } ~:r2
```

This is intended to be the same as the following N-Triples:

```turtle
:s :p :o .
:r1 rdf:reifies <<(:s :p :o)>> .
:r1 :p1 _:r0 .
_:r0 rdf:reifies <<(:s2 :p2 :o2)>> .
:r2 rdf:reifies <<(:s :p :o)>> .
```

However, because `curTripleTerm` is reset to `<<(:s2 :p2 :o2)>>` when parsing the inner reified triple, it forgets the original `<<(:s :p :o)>>` when it gets to `~:r2`. The same thing can happen with annotations. this would result in the following incorrect output:

```turtle
:s :p :o .
:r1 rdf:reifies <<(:s :p :o)>> .
:r1 :p1 _:r0 .
_:r0 rdf:reifies <<(:s2 :p2 :o2)>> .
:r2 rdf:reifies <<(:s2 :p2 :o2)>> .
```

Instead, productions need to save and restore the value of `curTripleTerm` on entry and exit to `reifiedTriple` and `annotationBlock ` productions.

### PR DESCRIPTION
**Pull request by gkellogg**
* Update Turtle grammar for Triple Term to be `<<(ttSubject verb ttObject)>>` instead of `<<(ttSubject predicate ttObject)>>`. This allows `a` in addition to an IRI.
* Save and restore curTripleTerm in Annotation Blocks and Reified Triples.
* Fix some HTML validator warnings.

Fixes #67.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-turtle/pull/68.html" title="Last updated on Sep 9, 2024, 6:51 PM UTC (9d69894)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-turtle/68/bc0650c...9d69894.html" title="Last updated on Sep 9, 2024, 6:51 PM UTC (9d69894)">Diff</a>

### CODE REVIEW

**Review by afs — APPROVED** [review]

**Review by domel — APPROVED** [review]

File: spec/index.html
Comment (TallTed) [inline spec/index.html:1863]:
```suggestion
      it next yields the RDF triple |curReifier| <code>rdf:reifies</code> |curTripleTerm|,
      and then restores the recorded value of the |curTripleTerm|.
```

File: spec/index.html
Comment (TallTed) [inline spec/index.html:1880]:
```suggestion
      If the |curReifier| is not set, then it is assigned a fresh RDF <a data-cite="RDF12-CONCEPTS#dfn-blank-node">blank node</a>
```

File: spec/index.html
Comment (TallTed) [inline spec/index.html:1883]:
```suggestion
      Finishing the <a href="#grammar-production-annotationBlock"><code>annotationBlock</code></a> production
```

File: spec/index.html
Comment (TallTed) [inline spec/index.html:1884]:
```suggestion
      clears the value of the |curReifier|
```

File: spec/index.html
Comment (TallTed) [inline spec/index.html:1887]:
```suggestion
      reifying triple |curReifier| <code>rdf:reifies</code> |curTripleTerm|
```

**Review by TallTed — CHANGES_REQUESTED** [review]
punctuation, typo, and language

**Comment by pchampin** [comment]
This was discussed during the [#rdf-star meeting on 12 September 2024](https://www.w3.org/2024/09/12-rdf-star-minutes.html#t07).

<details><summary><i>View the transcript</i></summary>

#### [w3c/rdf-turtle#68](https://github.com/w3c/rdf-turtle/issues/68) and [w3c/rdf-trig#33](https://github.com/w3c/rdf-trig/issues/33)
<cite>gkellogg:</cite> second item (updates to turtle). I think that's ready to merge.
  
  

----
</details>

**Comment by pchampin** [comment]
This was discussed during the [rdf-star meeting on 12 September 2024](https://www.w3.org/2024/09/12-rdf-star-minutes.html#t07).

<details><summary><i>View the transcript</i></summary>

#### [w3c/rdf-turtle#68](https://github.com/w3c/rdf-turtle/issues/68) and [w3c/rdf-trig#33](https://github.com/w3c/rdf-trig/issues/33)


<cite>gkellogg:</cite> second item (updates to turtle). I think that's ready to merge.
  
  


<cite>ora:</cite> you're saying someone else should take a look before a merge?


<cite>gkellogg:</cite> yes. look and approve. then we can merge after that.


<cite>AndyS:</cite> I can. Have done changes to the trig parser.


<cite>&lt;pchampin&gt;</cite> subtopic [w3c/sparql-federated-query#20](https://github.com/w3c/sparql-federated-query/issues/20)


<cite>ora:</cite> what about federated sparql error fix?


<cite>AndyS:</cite> it has 3 approvals.


<cite>ora:</cite> I think we should merge it. Not sensing that it needs waiting anymore.


<cite>AndyS:</cite> I can sort that out.


<cite>ora:</cite> any others for today?


----
</details>

**Pull request merged**
